Source/Eto/Forms/Notification.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using Eto.Drawing;
+
+namespace Eto.Forms
+{
+	/// <summary>
+	/// System toast notification.
+	/// </summary>
+	[Handler(typeof(Notification.IHandler))]
+	public class Notification : Widget
+	{
+		private Icon icon;
+
+		new IHandler Handler { get { return (IHandler)base.Handler; } }
+
+		static readonly object callback = new Callback();
+
+		/// <summary>
+		/// Gets an instance of an object used to perform callbacks to the widget from handler implementations.
+		/// </summary>
+		/// <returns>The callback instance to use for this widget.</returns>
+		protected override object GetCallback() { return callback; }
+
+		/// <summary>
+		/// Gets or sets the icon for the <see cref="Notification"/>.
+		/// </summary>
+		/// <remarks>
+		/// Does nothing on WPF and WinForms.
+		/// </remarks>
+		/// <value>The icon of the <see cref="Notification"/>.</value>
+		public Icon Icon
+		{
+			get { return icon; }
+			set
+			{
+				icon = value;
+				Handler.SetIcon(value);
+			}
+		}
+
+		/// <summary>
+		/// Gets or sets the message of the <see cref="Notification"/>.
+		/// </summary>
+		/// <value>The message of the <see cref="Notification"/>.</value>
+		public string Message
+		{
+			get { return Handler.Message; }
+			set { Handler.Message = value; }
+		}
+
+		/// <summary>
+		/// Gets or sets the title for the <see cref="Notification"/>.
+		/// </summary>
+		/// <value>The title for the <see cref="Notification"/>.</value>
+		public string Title
+		{
+			get { return Handler.Title; }
+			set { Handler.Title = value; }
+		}
+
+		/// <summary>
+		/// Shows the current notification.
+		/// </summary>
+		/// <remarks>
+		/// On some platforms like Gtk and Windows 10 the indicator
+		/// is not needed, while on the others like Windows 7 it's required.
+		/// </remarks>
+		/// <param name="indicator">Indicator to use to show the notification.</param>
+		public void Show(TrayIndicator indicator = null)
```

Comment:
Hm, how would one know they need a TrayIndicator here?  Perhaps we should add a `RequiresTrayIndicator` property of some sort so that you can create the TrayIndicator if it is required.